sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_async_client_manager.py

```diff
@@ -391,6 +437,11 @@ async def _setup_failover_endpoints(self):
                     )
         self._replica_clients = new_clients
         self._next_update_time = time.time() + MINIMAL_CLIENT_REFRESH_INTERVAL
+        if updated_endpoints and self._load_balancing_enabled:
+            # Reshuffle only if a new client was added and _load_balancing_enabled is enabled
+            # This allways needs to be shuffled, but if load balancing is enabled than the primary endpoint need to be
+            #  shuffled too.
+            random.shuffle(self._replica_clients)
```

Comment:
Shuffling logic needs to be updated here too. 
Also, the comment may not be valid anymore.